src/Sylius/Bundle/ApiBundle/Resources/config/validation/Customer.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+
+ This file is part of the Sylius package.
+
+ (c) Sylius Sp. z o.o.
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping"
+                    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                    xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping http://symfony.com/schema/dic/services/constraint-mapping-1.0.xsd">
+
+    <class name="Sylius\Component\Core\Model\Customer">
+        <property name="gender">
+            <constraint name="Choice">
+                <option name="choices">
+                    <value>u</value>
+                    <value>m</value>
+                    <value>f</value>
+                </option>
+                <option name="groups">
+                    <value>sylius_api_user_create</value>
+                </option>
+            </constraint>
+        </property>
+        <property name="user">
+            <constraint name="Valid" />
```

Comment:
Does it accept null value? I don't see mention of nullable behavior at https://symfony.com/doc/current/reference/constraints/Valid.html

Generally speaking let's cover this with group as this constrain becomes global sylius constraint otherwise.

Reply:
Yes, I wanted to add the `sylius_api` group, but I would also have to add the same group to the constraints on the `user` entity, because the `Valid` constraint passes on [the groups that we set](https://github.com/symfony/validator/blob/6.3/Constraints/ValidValidator.php). The solution to this problem would be to create a different constraint, as the person [here](https://github.com/symfony/symfony/issues/40741) suggestes. Then we could do it like this:
```
<property name="user">
    <constraint name="Valid">
        <option name="groups">
            <value>sylius_api</value>
        </option>
        <option name="embeddedGroups">
            <value>sylius_user_create</value>
            <value>sylius_user_update</value>
        </option>
    </constraint>
</property>
```